### PR TITLE
Wallet: Disable long-press feature on tokens in watch-only accounts. #18597

### LINKED ISSUE #18597: Wallet: Disable long-press feature on tokens in watch-only accounts.
# Feature Issue

This ticket is created since ticket #18543 is blocked. While waiting, the modal for the long-press in watch-only account should be disabled since it will cause the flow of the app to break.

Screenshot_2024-01-23_at_16 11 08

### PR DESCRIPTION
**Pull request by mmilad75**
fixes #18597 

https://github.com/status-im/status-mobile/assets/55688834/f7444f9a-5107-4558-b90c-5f8104d213e0

### CODE REVIEW

File: src/status_im/contexts/wallet/common/token_value/view.cljs [hunk 1]
```diff
@@ -41,10 +41,12 @@
...
-               :selected-item (fn [] [quo/token-value item])}])})])
+  (let [{:keys [watch-only?]} (rf/sub [:wallet/current-viewing-account])
+        on-long-press         (when (not watch-only?)
+                                #(rf/dispatch
+                                  [:show-bottom-sheet
+                                   {:content       (fn [] [token-value-drawer item])
+                                    :selected-item (fn [] [quo/token-value item])}]))]
+    [quo/token-value
+     (merge item
+            {:on-long-press on-long-press})]))
```
Comment (Parveshdhull) [inline src/status_im/contexts/wallet/common/token_value/view.cljs]: Here we can just use

```
cond-> item
(not watch-only?)
(assoc :on-long-press #(rf...))
```

**Review by vkjr — APPROVED** [review]

File: src/status_im/contexts/wallet/common/token_value/view.cljs (hunk 1 above)
Comment (mmilad75) [reply src/status_im/contexts/wallet/common/token_value/view.cljs]: Nice. Didn't know about this.

Comment (mmilad75) [reply src/status_im/contexts/wallet/common/token_value/view.cljs]: Done

**Review by J-Son89 — APPROVED** [review]

**Review by Parveshdhull — APPROVED** [review]

**Review by ulisesmac — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
@mmilad75 thank you for the fixes. PR is tested and ready to be merged

**Pull request merged**